### PR TITLE
SRCH-5170: Update Rails to 7.1.0

### PR DESCRIPTION
**Pull request by luisgmetzger**
## Summary
- This PR upgrades the Rails gem to version 7.1.0 based on this ticket: https://cm-jira.usa.gov/browse/SRCH-5170
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: config/initializers/new_framework_defaults_7_1.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # Be sure to restart your server when you modify this file.
 #
-# This file eases your Rails 7.0 framework defaults upgrade.
+# This file eases your Rails 7.1 framework defaults upgrade.
```
Comment (lorawoodford) [inline config/initializers/new_framework_defaults_7_1.rb:3]: Since we're already using all `7.1` defaults per the changes in `config/application.rb`, we don't need this file (and we didn't need the existing 7.0 one either but it snuck in there somehow).  https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#configure-framework-defaults

File: config/initializers/filter_parameter_logging.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Be sure to restart your server when you modify this file.
 
-# Configure sensitive parameters which will be filtered from the log file.
+# Configure parameters to be partially matched (e.g. passw matches password) and filtered from the log file.
+# Use this to limit dissemination of sensitive information.
+# See the ActiveSupport::ParameterFilter documentation for supported notations and behaviors.
 Rails.application.config.filter_parameters += [
-  :password, :passw, :secret, :token, :_key, :crypt, :salt, :certificate, :otp, :ssn, :query
+  :password, :passw, :secret, :token, :_key, :crypt, :salt, :certificate, :otp, :ssn
```
Comment (lorawoodford) [inline config/initializers/filter_parameter_logging.rb:7]: We intentionally include `query` here to keep from logging potentially sensitive query terms.  This needs to be added back.

File: spec/config/initializers/filter_parameter_logging_spec.rb [hunk 3]
```diff
@@ -1,12 +0,0 @@
-# frozen_string_literal: true
```
Comment (lorawoodford) [inline spec/config/initializers/filter_parameter_logging_spec.rb:1]: Again, it was an intentional decision to drop the query logging and this test is here to ensure that keeps happening.  It'll need to be added back.

**Review by lorawoodford — CHANGES_REQUESTED** [review]
@luisgmetzger Comments left in line.  Thanks!

**Review by lorawoodford — CHANGES_REQUESTED** [review]
@luisgmetzger Comments left in line.  Thanks!

**Comment by luisgmetzger** [comment]
@lorawoodford - I have addressed your comments. Ready for another review

File: spec/config/initializers/filter_parameter_logging_spec.rb [hunk 4]
```diff
@@ -7,6 +7,9 @@
   let(:parameter_filter) { ActiveSupport::ParameterFilter.new(config.filter_parameters) }
 
   it 'filters query from logs' do
-    expect(config.filter_parameters).to match(array_including(:query))
+    filter_contains_query = config.filter_parameters.any? do |param|
```
Comment (lorawoodford) [inline spec/config/initializers/filter_parameter_logging_spec.rb:10]: Why is this change necessary?  This seems to pass fine as is locally without this added complication.

**Review by lorawoodford — COMMENTED** [review]
@luisgmetzger Question in line.

File: spec/config/initializers/filter_parameter_logging_spec.rb (hunk 4 above)
Comment (luisgmetzger) [reply spec/config/initializers/filter_parameter_logging_spec.rb:10]: @lorawoodford  - I definitely am an agreement with you and have found that to be the case for me as well, locally ... but when running CI pipeline builds (with no change) it specifically fails on this spec.

Comment (luisgmetzger) [reply spec/config/initializers/filter_parameter_logging_spec.rb:10]: I ran it a few times with variations on the test implementation and the error message changes accordingly with it as well. This implementation did the trick and passed.

Comment (lorawoodford) [reply spec/config/initializers/filter_parameter_logging_spec.rb:10]: Ok, what about something like:

```
expect(config.filter_parameters.to_s).to match(/:query\b/)
```
That should match both locally when we get back:
```
[:password, :passw, :secret, :token, :_key, :crypt, :salt, :certificate, :otp, :ssn, :query]
```
and in CI when we (for whatever reason) get back:
```
[/(?i:password)|(?i:passw)|(?i:secret)|(?i:token)|(?i:_key)|(?i:crypt)|(?i:salt)|(?i:certificate)|(?i:otp)|(?i:ssn)|(?i:query)/]
```
I'm still a little curious/concerned why we're getting back different things in CI and locally... when we smoke test we'll definitely want to ensure this is still behaving as described in SRCH-3933.

**Comment by luisgmetzger** [comment]
@lorawoodford  - I have addressed your comment. Ready for another review.

**Review by lorawoodford — APPROVED** [review]
@luisgmetzger LGTM
